packages/ketcher-core/src/domain/entities/DrawingEntitiesManager.ts

```diff
@@ -463,4 +480,158 @@ export class DrawingEntitiesManager {
 
     return command;
   }
+
+  private rearrangeChain(initCoords: Vec2, chain, canvasWidth) {
+    const command = new Command();
+    const editorSettings = provideEditorSettings();
+    const monomerWidth = chain[0].renderer?.bodyWidth ?? 0;
+    const monomerHeight = chain[0].renderer?.bodyHeight ?? 0;
+    const heightMonomerWithBond =
+      monomerHeight + VERTICAL_DISTANCE_FROM_MONOMER;
+    chain[0].moveAbsolute(Scale.scaled2obj(initCoords, editorSettings));
+    const operation = new MonomerMoveOperation(chain[0]);
+    command.addOperation(operation);
+
+    for (let i = 1; i < chain.length; i++) {
+      const prevPosition = Scale.obj2scaled(
+        chain[i - 1].position,
+        editorSettings,
+      );
+      if (
+        prevPosition.x +
+          DISTANCE_BETWEEN_MONOMERS +
+          initCoords.x +
+          HORIZONTAL_DISTANCE_FROM_MONOMER +
+          DISTANCE_FROM_RIGHT >
+        canvasWidth
```

Comment:
I would propose to create a variable for that with a meaningful name

Reply:
Like isMonomerFitCanvas of something like that

Reply:
done